java_src/foam/nanos/auth/AuthOAO.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.oao.ProxyOAO;
+
+import javax.security.auth.AuthPermission;
+import java.util.Map;
+
+public class AuthOAO
+  extends ProxyOAO
+{
+  protected final String rootPermission_;
+
+  public AuthOAO() {
+    this.rootPermission_ = null;
+  }
+
+  public AuthOAO(String rootPermission_) {
+    this.rootPermission_ = rootPermission_;
+  }
+
+  protected String getRootPermission_(X x) throws IllegalAccessException {
+    return rootPermission_ == null ? getDelegate().get(x).getClassInfo().getId() : rootPermission_;
```

Comment:
This isn't going to work since rootPermission_ is final. Set it in the constructor. Have two constructors, one with args: (String rootPermission, OAO delegate) and one with (OAO delegate) and compute the default rootPermission in the second version.